libraries/base/ETA/JString.hs

```diff
@@ -28,12 +28,17 @@ import GHC.Real
 import Data.String
 import Data.Char
 import Foreign
+import GHC.Show (Show(..))
+import GHC.Pack (unpackCString)
 
 -- TODO: Add rules to simplify
 -- fromString (unpackCString# "Hello world!"#) :: JString = JString "Hello world"#
 instance IsString JString where
   fromString = mkJString
 
+instance Show JString where
+    show = unpackCString
```

Comment:
We can make it consistent by using `show . unpackCString`. But I don't have any strong opinions on it. I think making it equivalent to `Show String` would be nice as people may be relying on this behaviour:

```
show string == show jstring
```